src/TestSuite/Fixture/FixtureManager.php

```diff
@@ -227,14 +227,15 @@ protected function _loadFixtures($test)
     protected function _setupTable($fixture, $db, array $sources, $drop = true)
     {
         $configName = $db->configName();
-        if ($this->isFixtureSetup($configName, $fixture)) {
+        $isFixtureSetup = $this->isFixtureSetup($configName, $fixture);
+        if ($isFixtureSetup) {
             return;
         }
 
         $table = $fixture->sourceName();
         $exists = in_array($table, $sources);
 
-        if ($drop && $exists) {
+        if (($drop && $exists) || ($exists && !$isFixtureSetup)) {
```

Comment:
You're right both sides of the condition would be the same. Checking that the fixture has schema is a better approach.